### ISSUE TITLE
Puck2D custom heading

### ISSUE THREAD

**Issue by miacando** [mapbox/mapbox-maps-ios#2122]
I'm having trouble migrating this to v11:

In v10 I created a custom location provider that overrides the map location provider so that I can put a custom heading for the Puck2D:

```
public func locationManager(_ manager: CLLocationManager, didUpdateHeading heading: CLHeading) {
      let point1 = self.currentLocation
      let point2 = self.latestLocation
      let bearing = point1.direction(to: point2) + 180
      let heading = MyHeading()
      heading.trueHeading = bearing
      delegate?.locationProvider(self, didUpdateHeading: heading)
  }
```

This was for pointing the bearing to a certain direction.

However in v11, I could not use this custom location provider anymore. I could not find a similar implementation in the examples. Is there another way to point the Puck2D bearing to a certain direction or set a custom heading?

**Comment by aleksproger** [comment]
Hey, @miacando you may take a look at the [`HeadingProvider`](https://docs.mapbox.com/ios/maps/api/11.2.0-beta.1/documentation/mapboxmaps/headingprovider)
You can supply `MapView` with custom location and heading providers using [this method](https://docs.mapbox.com/ios/maps/api/11.2.0-beta.1/documentation/mapboxmaps/locationmanager/override(locationprovider:headingprovider:)-8xcsf)
The usage of custom location provider is also shown `NavigationSimulatorExample`

Basically, you may do something like this
```swift
mapView.location.override(locationProvider: customLocationProvider, headingProvider: customHeadingProvider)
```

Let me know, if you have any further questions or the solution works for you!

**Comment by miacando** [comment]
Thanks again @aleksproger, I was able to find a simple way to do it.
